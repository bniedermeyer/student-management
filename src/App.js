import React from 'react';
import { Link } from '@reach/router';
import { Pane, Text, Heading } from 'evergreen-ui';

import './App.css';

const NavLink = props => (
  <Link
    {...props}
    getProps={({ isCurrent }) => {
      return {
        style: {
          color: isCurrent ? 'red' : 'blue'
        }
      };
    }}
  />
);

function App(props) {
  return (
    <div className="App">
      <header>
        <nav>
          <Pane display="flex" padding={16} background="tint2" borderRadius={3}>
            <Pane flex={1} alignItems="center" display="flex">
              <Heading size={600}>Student Managemet</Heading>
            </Pane>
            <Pane>
              <NavLink to="/">
                <Text marginRight={16}>Home</Text>
              </NavLink>
              <NavLink to="/students">
                <Text marginRight={16}>Student Search</Text>
              </NavLink>
              <NavLink to="/general">
                <Text>General Search</Text>
              </NavLink>
            </Pane>
          </Pane>
        </nav>
      </header>
      {props.children}
    </div>
  );
}

export default App;
