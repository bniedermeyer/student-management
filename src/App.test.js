import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders the app', () => {
  const tree = renderer
    .create(
      <App>
        <div>My Other Components</div>
      </App>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
