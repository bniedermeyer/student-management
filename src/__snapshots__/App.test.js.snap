// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app 1`] = `
<div
  className="App"
>
  <header>
    <nav>
      <div
        className="css-nil ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦pb_16px ðŸ“¦pl_16px ðŸ“¦pr_16px ðŸ“¦pt_16px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦box-szg_border-box"
      >
        <div
          className="css-nil ðŸ“¦flx_1 ðŸ“¦algn-itms_center ðŸ“¦dspl_flex ðŸ“¦box-szg_border-box"
        >
          <h2
            className="ðŸ“¦mt_0px ðŸ“¦mb_0px ðŸ“¦fnt-sze_20px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_24px ðŸ“¦ltr-spc_-0-07px ðŸ“¦fnt-fam_wm9v6k ðŸ“¦color_234361 ðŸ“¦box-szg_border-box"
          >
            Student Management
          </h2>
        </div>
        <div
          className="css-nil ðŸ“¦box-szg_border-box"
        >
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "color": "red",
              }
            }
          >
            <span
              className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦mr_16px ðŸ“¦box-szg_border-box"
            >
              Home
            </span>
          </a>
          <a
            href="/students"
            onClick={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
          >
            <span
              className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦mr_16px ðŸ“¦box-szg_border-box"
            >
              Student Search
            </span>
          </a>
          <a
            href="/add-student"
            onClick={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
          >
            <span
              className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box"
            >
              Add Student
            </span>
          </a>
        </div>
      </div>
    </nav>
  </header>
  <div>
    My Other Components
  </div>
</div>
`;
